### PR TITLE
chore: nft slash

### PR DESCRIPTION
**Pull request by catngh**
**What does this PR do?**

- Remove author filter for pagination
- slash command for `$nft add / integrate / config twitter-sale`

### CODE REVIEW

File: src/commands/community/nft_slash/config.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      )
+      .addStringOption((option) =>
+        option
+          .setName("consumer_key_secret")
+          .setDescription("Your twitter consumer key secret. Example: hNl8...")
+          .setRequired(true)
+      )
+      .addStringOption((option) =>
+        option
+          .setName("access_token")
```
Comment (tuanddd) [inline src/commands/community/nft_slash/config.ts:30]: replace underscore with space

File: src/commands/community/nft_slash/config.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+          .setDescription("Your twitter access token. Example: 1450...")
+          .setRequired(true)
+      )
+      .addStringOption((option) =>
+        option
+          .setName("access_token_secret")
```
Comment (tuanddd) [inline src/commands/community/nft_slash/config.ts:36]: replace underscore with space

File: src/commands/community/nft_slash/config.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  },
+  run: async function (interaction: CommandInteraction) {
+    if (!interaction.guild || !interaction.guildId) {
+      throw new GuildIdNotFoundError({})
+    }
+
+    const csmrKey = interaction.options.getString("consumer_key")
+    const csmrKeyScrt = interaction.options.getString("consumer_key_secret")
+    const acsToken = interaction.options.getString("access_token")
+    const acsTokenScrt = interaction.options.getString("access_token_secret")
```
Comment (tuanddd) [inline src/commands/community/nft_slash/config.ts:49]: edit to match above and also define the string values upfront at the start of the file and use them below
e.g
```typescript
import .... from ...
import .... from ...

const CONSUMER_KEY = "consumer key"
const CONSUMER_KEY_SECRET = "consumer key secret"
...

// use them below

```

File: src/commands/community/nft_slash/config.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+    if (!csmrKey || !csmrKeyScrt || !acsToken || !acsTokenScrt) {
+      throw new CommandError({
+        description:
+          "Missing arguments. Please enter all 4 Twitter credentials",
+      })
+    }
+    console.log(csmrKey, csmrKeyScrt, acsToken, acsTokenScrt)
```
Comment (tuanddd) [inline src/commands/community/nft_slash/config.ts:57]: remove

File: src/commands/community/nft_slash/config.ts [hunk 5]
```diff
@@ -0,0 +1,77 @@
+import { CommandInteraction } from "discord.js"
+import { SlashCommandSubcommandBuilder } from "@discordjs/builders"
+import { CommandError, GuildIdNotFoundError } from "errors"
+import { composeEmbedMessage2 } from "utils/discordEmbed"
+import { SLASH_PREFIX } from "utils/constants"
+import { SlashCommand } from "types/common"
+import { handle } from "../nft/config"
+
+const command: SlashCommand = {
+  name: "config_twitter-sale",
```
Comment (tuanddd) [inline src/commands/community/nft_slash/config.ts:10]: not presentable to user, use space instead `config twitter sale`

File: src/commands/community/nft_slash/index.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+import integrate from "./integrate"
+import {
+  SlashCommandBuilder,
+  SlashCommandSubcommandBuilder,
+} from "@discordjs/builders"
+import { CommandInteraction } from "discord.js"
+
+const subCommands: Record<string, SlashCommand> = {
+  add,
+  "config_twitter-sale": config,
```
Comment (tuanddd) [inline src/commands/community/nft_slash/index.ts:15]: not presentable to user, use space instead `config twitter sale`

**Review by tuanddd — CHANGES_REQUESTED** [review]

File: src/commands/community/nft_slash/config.ts [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+  category: "Community",
+  prepare: () => {
+    return new SlashCommandSubcommandBuilder()
+      .setName("config_twitter-sale")
```
[the remark this answers is not in the document]
Comment (catngh) [reply src/commands/community/nft_slash/config.ts:14]: discord doesn't allow names with spaces, i will make it a subcommand

File: src/utils/discordEmbed.ts [hunk 8]
```diff
@@ -541,7 +541,6 @@ export function listenForPaginateAction(
     .createMessageComponentCollector({
       componentType: MessageComponentTypes.BUTTON,
       idle: 60000,
-      filter: authorFilter(replyMsg.author.id),
```
Comment (hoangharry) [inline src/utils/discordEmbed.ts:544]: cần remove cái này hả e?

**Review by tuanddd — APPROVED** [review]

File: src/utils/discordEmbed.ts (hunk 8 above)
Comment (catngh) [reply src/utils/discordEmbed.ts:544]: dạ nó làm bug feature qua trang, vẫn work trên button thường

**Review by hoangharry — APPROVED** [review]

**Pull request merged**